src/internal/TLC5947GenericMethod.h

```diff
@@ -0,0 +1,253 @@
+
+#if !defined(__AVR_ATtiny85__) && !defined(ARDUINO_attiny)
+    TLC5947MethodBase(uint8_t pinLatch, uint8_t pinOutputEnable, uint16_t pixelCount, size_t elementSize, size_t settingsSize) :
+        TLC5947MethodBase(SCK, MOSI, pinLatch, pinOutputEnable, pixelCount, elementSize, settingsSize)
+    {
+    }
+
+    TLC5947MethodBase(uint8_t pinLatch, uint16_t pixelCount, size_t elementSize, size_t settingsSize) :
+        TLC5947MethodBase(SCK, MOSI, pinLatch, -1, pixelCount, elementSize, settingsSize)
+    {
+    }
+#endif
+
+    ~TLC5947MethodBase()
+    {
+        free(_data);
+        pinMode(_pinLatch, INPUT);
+        if (_pinOutputEnable >= 0)
+            pinMode(_pinOutputEnable, INPUT);
+    }
+
+    bool IsReadyToUpdate() const
+    {
+        return true; // dot stars don't have a required delay
+    }
+
+#if defined(ARDUINO_ARCH_ESP32)
+    void Initialize(int8_t sck, int8_t miso, int8_t mosi, int8_t ss)
+    {
+        _wire.begin(sck, miso, mosi, ss);
+    }
+#endif
+
+    void Initialize()
+    {
+        _wire.begin();
+    }
+
+    void WriteBuffer(uint8_t* buffer)
+    {
+        if (_pinOutputEnable >= 0)
+            digitalWrite(_pinOutputEnable, HIGH);
+        
+        digitalWrite(_pinLatch, LOW);
+        _wire.beginTransaction();
+
+       
+        _wire.transmitBytes(buffer, 36);   
+      
+        _wire.endTransaction();
+        digitalWrite(_pinLatch, HIGH);
+        digitalWrite(_pinLatch, LOW);
+        if (_pinOutputEnable >= 0)
+            digitalWrite(_pinOutputEnable, LOW);
+    }
+
+    void Update(bool)
+    { 
+        
+        if (_pinOutputEnable >= 0)
+            digitalWrite(_pinOutputEnable, HIGH);
+        
+        digitalWrite(_pinLatch, LOW);
+        _wire.beginTransaction();
+
+        uint8_t* ptr = _data + ((_channelCount - 1) * T_BITCONVERT::ChannelSize);
+        uint16_t remCh = _channelCount;
+        uint16_t numCh = _channelCount % 24;
+        if (numCh == 0)
+            numCh = 24;
+
+        while (remCh > 0)
+        {
+            T_BITCONVERT::ConvertFrame(_frame, ptr, numCh);
+            _wire.transmitBytes(_frame, sizeof(_frame));
+            ptr -= numCh * T_BITCONVERT::ChannelSize;
+            remCh -= numCh;
+            numCh = 24;
+        }           
+      
+        _wire.endTransaction();
+        digitalWrite(_pinLatch, HIGH);
+        digitalWrite(_pinLatch, LOW);
+        if (_pinOutputEnable >= 0)
+            digitalWrite(_pinOutputEnable, LOW);
+    }
+
+    uint8_t* getData() const
+    {
+        return _data;
+    };
+
+    size_t getDataSize() const
+    {
+        return _sizeData;
+    };
+
+    void applySettings(const SettingsObject& settings)
+    {
+        _wire.applySettings(settings);
+    }
+
+private:
+    const size_t   _sizeData;   // Size of '_data' buffer below
+    const uint16_t _channelCount;
+
+    T_TWOWIRE _wire;
+    uint8_t*  _data;       // Holds LED color values
+    uint8_t   _frame[36];  // Holds channel values for one module
```

Comment:
and there is that magic number 36 again ;-)

`const uint16_t sizeModuleChannel = 36`